umbraco/Umbraco-CMS
# Allow indexing variant nodes when not all variants are published - fixes issues 11383.

### Description

This fixes https://github.com/umbraco/Umbraco-CMS/issues/11383

### The fix

The introduction of the new [Examine library changes](https://github.com/umbraco/Umbraco-CMS/commit/eba6373a1239d89770fa02287665b7ebfa3bd5db) caused an unintended consequence that it stopped indexing of nodes with language variants on them when one of the variants was unpublished.

It did this because meaning of ValueSetValidationStatus.Filtered became overloaded as:

1) This node is to be filtered from the search result - for instance if it is in the recycle bin
2) The set of values under this node have been exluded - for instance if one variant of the node is unpublished

The umbraco content index functionality uses "filtered" to decide whether to delete the node - i.e. inline with scenario 1. Unfortunately it means any node with an unpublished variant also gets caught in the net.

These changes align ValueSetValidationStatus.Filtered to indicate that a node is intended as filtered out of a search, not that parts of its contents had been excluded from the result.

This brings it inline with how it is used in Umbraco.Examine.Lucene/UmbracoContentIndex

Unit tests changed to indicate the intent of ValueSetValidationStatus.Filtered

Change to UmbracoViewPage to make model variable nullable (because the solution wouldn't build otherwise on 2022)

### Steps to test

1) Create a new umbraco system

2) Install the [detault starter kit](https://our.umbraco.com/packages/starter-kits/the-starter-kit/)

3) Follow the instructions for how to [implement Examine](https://our.umbraco.com/documentation/Reference/Searching/Examine/quick-start/)  

4) Make like easier by allowing your newly implemented search to search all content by changing the searcher code and removing the NodeTypeAlias("person") contrainst e.g.

    var searcher = index.Searcher;
    var results = searcher.CreateQuery("content").Field("nodeName", searchTerm).Execute();

5) If you do a search for say "contact" it should return the contact page:

![image](https://user-images.githubusercontent.com/2386725/177777141-16921fe9-ca11-4e6a-bd10-1d77fa43351e.png)

6) Follow the instruction for how to [make your site multilingual](https://our.umbraco.com/documentation/tutorials/multilanguage-setup/)

7)  Make a the contact node multi-lingual - put when you go publish it - only publish the root language.
![image](https://user-images.githubusercontent.com/2386725/177777482-793f91c7-c35b-4954-821b-5b0640d0bae7.png)

8) Repeat you search - before the fix, you will get no responses. After the fix the contact page will still be present.

Note: The fix applies at time of indexing, not at the searching. I.e. if you published your pages before the fix - you will need to re-publish after the fix to get your pages indexed.

### Unintended behaviours

This is correcting the behaviour of  how "filtered" on ValueSetValidationStatus is interpreted / making it consistent. 

I have done searches through the code and the status is not used in any other functionality so it looks safe. Existing unit tests back this up also. 

The ValueSetValidator class is public, so it is possible that external code could be using the filter status in what will now be the incorrect way (to indicate an exclusion of innards), however since the change was only introduced relatively recently I think it is unlikely. None the less - it could be so it may be worth mentioning in the release notes as a potential change in behaviour?     

### Prerequisites
